Tasks/Maven/maventask.ts

```diff
@@ -284,12 +287,28 @@ function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
-// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function cleanDirectory(targetDirectory:string):Q.Promise<void> {
+    var defer = Q.defer<void>();
+
+    tl.rmRF(targetDirectory);
+    tl.mkdirP(targetDirectory);
+
+    if (tl.exist(targetDirectory)) {
+        defer.resolve(undefined);
+    } else {
+        tl.debug('Failed to clean ' + targetDirectory + ': Could not create new directory after recursive delete.');
+        defer.reject(undefined);
+    }
+
+    return <Q.Promise<void>>defer.promise;
+}
+
+// Take a result object and promise to upload the XML and HTML reports as artifacts. Throws if fail.
 function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
-    // make sure the staging folder exists and is empty
-    tl.rmRF(stagingDir);
-    tl.mkdirP(stagingDir);
+    var toolStagingDir:string = path.join(stagingDir, '.' + analysisResult.toolName.toLowerCase());
```

Comment:
P3: Too many dots in the directory names, I would keep these without it since the parent directory has .